core/src/test/scala/io/delta/tables/DeltaTableBuilderSuite.scala

```diff
@@ -228,6 +228,18 @@ class DeltaTableBuilderSuite extends QueryTest with SharedSparkSession with Delt
     }
   }
 
+  test("test addColumn using columnBuilder, without dataType") {
+    val e = intercept[AnalysisException] {
+      io.delta.tables.DeltaTable.create().addColumn(
```

Comment:
we can just run 

```
DeltaTable.columnBuilder("value")
          .generatedAlwaysAs("true")
          .nullable(true)
          .build()
```

This would show that we will catch the error in `build()`.

Reply:
got it , makes sense !